providers/common/provider_util.c

```diff
@@ -89,18 +91,12 @@ ENGINE *ossl_prov_cipher_engine(const PROV_CIPHER *pc)
     return pc->engine;
 }
 
-void ossl_prov_cipher_clear(PROV_CIPHER *pc)
-{
-    EVP_CIPHER_free(pc->alloc_cipher);
-    pc->alloc_cipher = NULL;
-    pc->cipher = NULL;
-    pc->engine = NULL;
-}
-
 void ossl_prov_digest_free(PROV_DIGEST *pd)
```

Comment:
Nit on the names...  considering this doesn't actually *free* the `PROV_DIGEST`, only its fields when applicable, I think `ossl_prov_digest_clear` is the better name.  This, so no one gets the wrong idea down the line...

Do you agree?

Reply:
I agree, it could be misleading.

I'm not sure about the `ossl_prov_digest_clear` name, since it is doing more than that.
`ossl_prov_digest_release`?

Reply:
I dunno...  if you have a function that clears a structure, isn't it sensible that if some field needs freeing, that's what's going to happen?

...

but yeahok, I can go with `release` as well.  Just not `free` :wink:

Reply:
Release still indicates that the pd is released. I think clear is fine. If you do not like it how about `ossl_prov_digest_reset`?

Reply:
`reset` works too, and we have a precedent too.

Reply:
I like _reset.  I'll change to that.